code/pages/ExtensibleSearchPage.php

```diff
@@ -719,11 +748,76 @@ public function getSearchResults($data = null, $form = null) {
 
 		$startTime = microtime(true);
 
+		if($results = $this->getSearchEngineResults($data, $form)) {
```

Comment:
I feel like this needs some revision..

If your search engine is set to full-text, this will cause the get search engine results to return null, which triggers the full-text search below. The full-text search should actually be encapsulated in the get search engine results function. This is likely the result of poor commenting, which mentions it being a fallback to begin with :)

This would also raise the question, should there be a full-text fallback if the current search engine fails to return any results? I would be leaning towards yes, considering you want to try and find *something* for the user, however this means they could be waiting twice as long for no results.